### PR TITLE
add cron schedule and workflow id reuse policy

### PR DESCRIPTION
**Pull request by duoertai**

### CODE REVIEW

File: service/api/cadence/client.go [hunk 1]
```diff
@@ -143,6 +149,21 @@ func mapToIwfSearchAttributes(searchAttributes *shared.SearchAttributes) (map[st
...
+	case service.WorkflowIDReusePolicyAllowDuplicate:
+		return client.WorkflowIDReusePolicyAllowDuplicate, nil
+	case service.WorkflowIDReusePolicyAllowDuplicateFailedOnly:
+		return client.WorkflowIDReusePolicyAllowDuplicateFailedOnly, nil
+	case service.WorkflowIDReusePolicyRejectDuplicate:
+		return client.WorkflowIDReusePolicyRejectDuplicate, nil
+	case service.WorkflowIDReusePolicyTerminateIfRunning:
+		return client.WorkflowIDReusePolicyTerminateIfRunning, nil
+	default:
+		return client.WorkflowIDReusePolicyAllowDuplicate, fmt.Errorf("unsupported workflow id reuse policy %s", workflowIdReusePolicy)
```
Comment (duoertai) [inline service/api/cadence/client.go:163]: cadence doesn't have a UNSPECIFIED enum

File: service/api/cadence/client.go [hunk 2]
```diff
@@ -36,10 +36,16 @@ func (t *cadenceClient) Close() {
...
+	workflowIdReusePolicy, err := mapToTemporalWorkflowIdReusePolicy(options.WorkflowIDReusePolicy)
+	if err != nil {
+		return "", nil
+	}
 	workflowOptions := client.StartWorkflowOptions{
 		ID:                           options.ID,
 		TaskList:                     options.TaskQueue,
 		ExecutionStartToCloseTimeout: options.WorkflowRunTimeout,
+		WorkflowIDReusePolicy:        workflowIdReusePolicy,
+		CronSchedule:                 options.CronSchedule,
```
Comment (duoertai) [inline service/api/cadence/client.go:48]: @longquanzheng do we need validate cron schedule here?

Comment (longquanzheng) [reply service/api/cadence/client.go:48]: I think the Cadence/Temporal go SDK will do it so we don't need to do here

File: service/api/cadence/client.go [hunk 3]
```diff
@@ -143,6 +149,21 @@ func mapToIwfSearchAttributes(searchAttributes *shared.SearchAttributes) (map[st
 	return result, nil
 }
 
+func mapToTemporalWorkflowIdReusePolicy(workflowIdReusePolicy string) (client.WorkflowIDReusePolicy, error) {
```
Comment (longquanzheng) [inline service/api/cadence/client.go:152]: nit: `mapToCadenceWorkflowIdReusePolicy`

File: service/api/cadence/client.go (hunk 1 above)
Comment (longquanzheng) [reply service/api/cadence/client.go:163]: yeah returnin gerror should be fine

Comment (longquanzheng) [reply service/api/cadence/client.go:163]: we probably should return nil for the policy (need to change the signature to use pointer)

**Pull request merged**